processor/src/operations/io_ops.rs

```diff
@@ -75,6 +80,44 @@ impl Process {
         Ok(())
     }
 
+    /// Loads two words from memory and adds their contents to the top 8 elements of the stack.
+    ///
+    /// The operation works as follows:
+    /// - The memory address of the first word is retrieved from 13th stack element (position 12).
+    /// - Two consecutive words, starting at this address, are loaded from memory.
+    /// - Elements of these words are added to the top 8 elements of the stack (element-wise, in
+    ///   stack order).
+    /// - Memory address (in position 12) is incremented by 2.
+    /// - All other stack elements remain the same.
+    pub(super) fn op_mstream(&mut self) -> Result<(), ExecutionError> {
```

Comment:
Of course not a blocker, but it seems that this op is but a script of other ops (load, add, inc).

wouldn't that suggest this instead should be implemented as an optional helper library (maybe in stdlib?) instead of adding a new atomic instruction? this would be aligned with the idea that the assembly language contains only the atomic building blocks, and the library implementations will contain aux routines that will support the user to implement their own routines - ranging from crypto primitives to full applications

Reply:
Actually, many (most?) assembly instructions at translated into short sequences of atomic operations. We call these "macro instructions". So, Miden assemble is just an abstraction level over the native Miden VM instruction set. In Miden assembly we currently have 200+ instructions, but there are only 84 instructions in the VM instruction set.